xc-admin/packages/xc-admin-common/src/propose.ts

```diff
@@ -0,0 +1,176 @@
+import Squads, { getIxAuthorityPDA, getTxPDA } from "@sqds/mesh";
+import {
+  PublicKey,
+  Transaction,
+  TransactionInstruction,
+} from "@solana/web3.js";
+import { BN } from "bn.js";
+import { AnchorProvider } from "@project-serum/anchor";
+import {
+  createWormholeProgramInterface,
+  getPostMessageAccounts,
+} from "@certusone/wormhole-sdk/lib/cjs/solana/wormhole";
+import { encodeExecutePostedVaa } from "./governance_payload/ExecutePostedVaa";
+
+type SquadInstruction = {
+  instruction: TransactionInstruction;
+  authorityIndex?: number;
+  authorityBump?: number;
+  authorityType?: string;
+};
+
+/**
+ * Propose an array of `TransactionInstructions` as a proposal
+ * @param squad Squads client
+ * @param vault vault public key (the id of the multisig where these instructions should be proposed)
+ * @param instructions instructions that will be proposed
+ * @param remote whether the instructions should be executed in the chain of the multisig or remotely on Pythnet
+ * @returns the newly created proposal's pubkey
+ */
+export async function proposeInstructions(
+  squad: Squads,
+  vault: PublicKey,
+  instructions: TransactionInstruction[],
+  remote: boolean,
+  wormholeAddress?: PublicKey
```

Comment:
make a class like `MultisigWrapper` to encapsulate some of the common arguments (squads, wormholeAddress, etc)